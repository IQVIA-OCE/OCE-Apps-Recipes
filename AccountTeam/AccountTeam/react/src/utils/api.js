import { sfNetAPI } from 'oce-apps-bridges';

export default {
  query: async (soql = '') =>
    new Promise((resolve, reject) => {
      sfNetAPI.query(
        soql,
        data => {
          if (data && data.records) {
            const { records, ...metadata } = data;
            resolve([records, metadata]);
          } else {
            resolve(data);
          }
        },
        e => {
          reject(e);
        }
      );
    }),
  describe: async (objectType = '') =>
    new Promise((resolve, reject) => {
      sfNetAPI.describe(
        objectType,
        data => {
          resolve([data]);
        },
        e => {
          reject(e);
        }
      );
    })
};
