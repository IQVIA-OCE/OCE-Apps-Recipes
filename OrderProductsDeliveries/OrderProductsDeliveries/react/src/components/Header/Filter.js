import React from "react";
import { View, StyleSheet } from "react-native";
import { Select } from 'apollo-react-native';
import { useSelector, useDispatch } from "react-redux";
import { setBrandFilterValue } from "../../stores/orders";
import {brandFilterSelector, brandOptionsSelector} from "../../stores/ordersSelectors";

export default () => {

    const dispatch = useDispatch();
    const brandsOptions = useSelector(brandOptionsSelector);
    const brandFilterValue = useSelector(brandFilterSelector);

    return (
        <View style={styles.content} testID="filterWrapper">
            <Select
                options={brandsOptions}
                value={brandFilterValue}
                onChange={val => dispatch(setBrandFilterValue(val?.value === 'none' ? null : val))}
                placeholder="Brand"
                width={'100%'}
                disabled={brandsOptions.length ? false : true}
                hideDropdownPlaceholder={true}
            />
        </View>
    );
};

const styles = StyleSheet.create({
    content: {
        width: 300
    },
});
