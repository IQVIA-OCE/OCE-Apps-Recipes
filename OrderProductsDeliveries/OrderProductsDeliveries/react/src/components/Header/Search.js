import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Search } from '@oce-apps/apollo-react-native';
import { useSelector, useDispatch } from 'react-redux';
import { setSearchProductValue } from '../../stores/orders';
import {searchProductValueSelector} from "../../stores/ordersSelectors";

export default () => {
  const dispatch = useDispatch();
  const searchProductValue = useSelector(searchProductValueSelector);
  return (
    <View style={styles.content}>
      <Search
        testID="search-view"
        placeholder="Search product name"
        value={searchProductValue}
        onChangeText={(text) => dispatch(setSearchProductValue(text))}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  content: {
    width: 300,
  },
});
