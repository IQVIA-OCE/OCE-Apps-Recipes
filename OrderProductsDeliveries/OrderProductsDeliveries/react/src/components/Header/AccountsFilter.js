import React, { useState, useEffect } from 'react';
import { Text, View, ScrollView, StyleSheet, Keyboard, Platform } from 'react-native';
import { Search, Menu, ApolloProgress, useTheme } from 'apollo-react-native';
import { fetchAccounts } from './../../api/OrderDetails';
import { localized, environment } from 'oce-apps-bridges';
import { setRecordId, setOrderDeliveries, setAccountFilterVisible } from '../../stores/orders';
import useDebounce from '../../utils/useDebounce';
import { useDispatch } from 'react-redux';

const AccountsFilter = () => {
  const dispatch = useDispatch();
  const NAMESPACE = environment.namespace();
  const [accountsList, setAccountsList] = useState([]);
  const [searchValue, setSearchValue] = useState('');
  const [isMenuVisible, setMenuVisible] = useState(false);
  const [isListLoading, setListLoading] = useState(false);

  const theme = useTheme();

  const debouncedSearch = useDebounce(searchValue, 500);

  useEffect(() => {
    fetchAccountsList(searchValue);
  }, [debouncedSearch]);

  const fetchAccountsList = async (value) => {
    try {
      setListLoading(true);
      const { records } = await fetchAccounts(value.trim());
      const normalizedAccounts = records.map((el) => {
        return {
          label: el.Name,
          value: el.Id,
        };
      });
      setAccountsList(normalizedAccounts);
      setListLoading(false);
    } catch (e) {
      console.warn({ e });
    }
  };

  return (
    <Menu
      visible={isMenuVisible}
      onDismiss={() => setMenuVisible(false)}
      testID="menu"
      anchor={
        <Search
          testID="search"
          placeholder={localized(`${NAMESPACE}account_name`, 'Account Name')}
          onFocus={() => setMenuVisible(true)}
          onChangeText={(value) => {
            if (value && Platform.OS !== 'web') setMenuVisible(true);
            setSearchValue(value);
          }}
          onSubmitEditing={Keyboard.dismiss}
          onIconPress={async () => {
            dispatch(setRecordId(null));
            setSearchValue('');
            dispatch(setOrderDeliveries([]));
            Keyboard.dismiss();
          }}
          value={searchValue}
          style={{ width: 300 }}
        />
      }
    >
      {isListLoading && (
        <View style={[styles.listLoader, { backgroundColor: theme.colors.surface }]} testID="loading">
          <ApolloProgress />
        </View>
      )}
      {accountsList.length === 0 && (
        <Text style={{ textAlign: 'center', paddingVertical: 20 }} testID="noelement">
          No accounts found
        </Text>
      )}
      <ScrollView
        style={{ maxHeight: 300, width: 295 }}
        keyboardShouldPersistTaps="always"
        testID="accountList"
      >
        {accountsList.map((el) => {
          return (
            <Menu.Item
              key={el.value}
              title={el.label}
              onPress={() => {
                dispatch(setRecordId(el.value));
                dispatch(setAccountFilterVisible(true));
                setSearchValue(el.label);
                setMenuVisible(false);
                Keyboard.dismiss();
              }}
            />
          );
        })}
      </ScrollView>
    </Menu>
  );
};

const styles = StyleSheet.create({
  content: {
    width: 300,
  },
  listLoader: {
    position: 'absolute',
    top: 0,
    left: 0,
    width: '100%',
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(255, 255, 255, 0.8)',
    zIndex: 2,
  },
});

export default AccountsFilter;
