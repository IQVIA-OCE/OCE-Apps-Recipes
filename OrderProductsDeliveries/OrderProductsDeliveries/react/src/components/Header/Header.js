import React from 'react';
import { View, StyleSheet } from 'react-native';
import Filter from './Filter';
import Search from './Search';
import AccountsFilter from './AccountsFilter';
import { Headline, useTheme } from '@oce-apps/apollo-react-native';
import { isAccountFilterVisibleSelector } from '../../stores/ordersSelectors';
import { useSelector } from 'react-redux';

export const Header = () => {
  const theme = useTheme();
  const backgroundColor = theme.dark ? '#2c2c2f' : '#fff';
  const isAccountFilterVisible = useSelector(isAccountFilterVisibleSelector);
  return (
    <View style={{ paddingBottom: 20, backgroundColor }}>
      <Headline>Deliveries</Headline>
      <View style={styles.settingsContainer}>
      {isAccountFilterVisible && <AccountsFilter />}
        <Search />
        <Filter />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  settingsContainer: {
    paddingTop: 15,
    flexDirection: 'row',
    justifyContent: 'space-between',
    width: '100%',
  },
});
