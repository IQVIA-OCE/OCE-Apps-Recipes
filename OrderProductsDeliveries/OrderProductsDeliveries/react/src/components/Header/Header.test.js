import { render, waitFor, act, fireEvent } from '@testing-library/react-native';
import React from 'react';
import { Header } from './Header';
import { useSelector, useDispatch } from 'react-redux';
import { fetchAccounts } from '../../api/OrderDetails';
import useDebounce from '../../utils/useDebounce';
import { Provider } from 'apollo-react-native';

jest.mock('react-redux', () => ({
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock('./../../api/OrderDetails', () => ({
    fetchAccounts: jest.fn(),
}));

jest.mock('../../utils/useDebounce');

const accounts = [
    {
        Id: '123',
        Name: '321'
    }
]

beforeEach(() => {
    jest.useFakeTimers({ doNotFake: ['nextTick', 'setImmediate'] });
})

afterEach(() => {
    jest.useRealTimers();
});

it('Header with accounts filter', async () => {
    useDebounce.mockImplementation((value) => value);
    fetchAccounts.mockResolvedValue({
        records: accounts
    })
    useDispatch.mockReturnValue(() => jest.fn());
    useSelector.mockImplementation((cb) => cb({
        orders: {
            _recordId: null,
            searchProductValue: '',
            brandOptions: [],
            isAccountFilterVisible: true
        }
    }));
    const { getByTestId } = render(
        <Provider>
            <Header />
        </Provider>
    );
    const search = getByTestId('search');
    fireEvent(search, 'focus');
    await waitFor(() => {
        expect(getByTestId('accountList')).toBeTruthy();
    });
});
