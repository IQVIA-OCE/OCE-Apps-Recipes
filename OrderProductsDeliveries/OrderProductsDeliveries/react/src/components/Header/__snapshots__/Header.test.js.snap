// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 1`] = `
<View
  style={
    Object {
      "marginBottom": 20,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
              "letterSpacing": 0,
              "lineHeight": 32,
              "marginVertical": 2,
            },
            undefined,
          ],
          Array [
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
              "letterSpacing": 0,
              "lineHeight": 32,
              "marginVertical": 2,
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Deliveries
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingTop": 15,
        "width": "100%",
      }
    }
  >
    <View
      collapsable={false}
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "width": 176,
            },
            Array [
              Object {
                "width": "100%",
              },
              undefined,
              Object {
                "width": 300,
              },
            ],
            Object {
              "opacity": 1,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#D9D9D9",
                "borderRadius": 4,
                "borderWidth": 1,
              },
              Object {
                "width": 300,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
                "minHeight": 40,
                "overflow": "hidden",
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          >
            <View
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "left": 10,
                  "opacity": 1,
                  "position": "absolute",
                  "right": "auto",
                  "top": "50%",
                  "transform": Array [
                    Object {
                      "translateY": -12.5,
                    },
                  ],
                  "zIndex": 5,
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#999",
                      "fontSize": 25,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <TextInput
              editable={true}
              maxResizableWidth={750}
              minHeight={60}
              minWidth={0}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              paddingLeft={40}
              paddingRight={8}
              placeholder="Account Name"
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              selectionColor="#0768FD"
              style={
                Array [
                  Object {
                    "margin": 0,
                    "textAlign": "left",
                    "zIndex": 5,
                  },
                  undefined,
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "height": 40,
                    "minHeight": 0,
                    "paddingLeft": 40,
                    "paddingRight": 8,
                    "paddingVertical": 5,
                    "textAlignVertical": "center",
                    "width": "100%",
                  },
                ]
              }
              testID="search"
              value=""
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "width": 300,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "width": 176,
            },
            Array [
              Object {
                "width": "100%",
              },
              undefined,
              undefined,
            ],
            Object {
              "opacity": 1,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#D9D9D9",
                "borderRadius": 4,
                "borderWidth": 1,
              },
              Object {
                "width": "100%",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
                "minHeight": 40,
                "overflow": "hidden",
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          >
            <View
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "left": 10,
                  "opacity": 1,
                  "position": "absolute",
                  "right": "auto",
                  "top": "50%",
                  "transform": Array [
                    Object {
                      "translateY": -12.5,
                    },
                  ],
                  "zIndex": 5,
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#999",
                      "fontSize": 25,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <TextInput
              editable={true}
              maxResizableWidth={750}
              minHeight={60}
              minWidth={0}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              paddingLeft={40}
              paddingRight={8}
              placeholder="Search product name"
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              selectionColor="#0768FD"
              style={
                Array [
                  Object {
                    "margin": 0,
                    "textAlign": "left",
                    "zIndex": 5,
                  },
                  undefined,
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "height": 40,
                    "minHeight": 0,
                    "paddingLeft": 40,
                    "paddingRight": 8,
                    "paddingVertical": 5,
                    "textAlignVertical": "center",
                    "width": "100%",
                  },
                ]
              }
              testID="search-view"
              value=""
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "width": 300,
        }
      }
      testID="filterWrapper"
    >
      <View
        style={
          Array [
            Object {
              "width": "100%",
            },
            undefined,
          ]
        }
      >
        <View>
          <View
            collapsable={false}
          >
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Object {
                    "backgroundColor": "rgba(246, 246, 246, 0.4)",
                    "borderColor": "rgba(217, 217, 217, 0.4)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "elevation": 0,
                    "height": 40,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "stretch",
                    "flex": 1,
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "paddingLeft": 8,
                  }
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  hitSlop={
                    Object {
                      "bottom": 14,
                      "left": 14,
                      "right": 14,
                      "top": 14,
                    }
                  }
                  nativeID="animatedComponent"
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "color": "rgba(0, 0, 0, 0.4)",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": "400",
                        },
                      ]
                    }
                  >
                    Brand
                  </Text>
                </View>
              </View>
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": true,
                  }
                }
                accessible={true}
                centered={true}
                collapsable={false}
                focusable={true}
                hitSlop={
                  Object {
                    "bottom": 6,
                    "left": 6,
                    "right": 6,
                    "top": 6,
                  }
                }
                nativeID="animatedComponent"
                onClick={[Function]}
                onLayout={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="box-only"
                style={Object {}}
              >
                <View
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "margin": 6,
                          "overflow": "hidden",
                        },
                        Object {
                          "borderRadius": 18,
                          "height": 36,
                          "width": 36,
                        },
                        Object {
                          "opacity": 0.32,
                        },
                        Object {
                          "margin": 0,
                        },
                      ],
                    ]
                  }
                >
                  <View>
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      style={
                        Array [
                          Object {
                            "color": "rgba(0, 0, 0, 0.4)",
                            "fontSize": 24,
                          },
                          Array [
                            Object {
                              "transform": Array [
                                Object {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                          ],
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 0,
                        "left": 0,
                        "overflow": "hidden",
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "borderRadius": 0,
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Header with acoounts filter 1`] = `
<View
  style={
    Object {
      "marginBottom": 20,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
              "letterSpacing": 0,
              "lineHeight": 32,
              "marginVertical": 2,
            },
            undefined,
          ],
          Array [
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
              "letterSpacing": 0,
              "lineHeight": 32,
              "marginVertical": 2,
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Deliveries
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingTop": 15,
        "width": "100%",
      }
    }
  >
    <View
      collapsable={false}
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "width": 176,
            },
            Array [
              Object {
                "width": "100%",
              },
              undefined,
              Object {
                "width": 300,
              },
            ],
            Object {
              "opacity": 1,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#D9D9D9",
                "borderRadius": 4,
                "borderWidth": 1,
              },
              Object {
                "width": 300,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
                "minHeight": 40,
                "overflow": "hidden",
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          >
            <View
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "left": 10,
                  "opacity": 1,
                  "position": "absolute",
                  "right": "auto",
                  "top": "50%",
                  "transform": Array [
                    Object {
                      "translateY": -12.5,
                    },
                  ],
                  "zIndex": 5,
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#999",
                      "fontSize": 25,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <TextInput
              editable={true}
              maxResizableWidth={750}
              minHeight={60}
              minWidth={0}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              paddingLeft={40}
              paddingRight={8}
              placeholder="Account Name"
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              selectionColor="#0768FD"
              style={
                Array [
                  Object {
                    "margin": 0,
                    "textAlign": "left",
                    "zIndex": 5,
                  },
                  undefined,
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "height": 40,
                    "minHeight": 0,
                    "paddingLeft": 40,
                    "paddingRight": 8,
                    "paddingVertical": 5,
                    "textAlignVertical": "center",
                    "width": "100%",
                  },
                ]
              }
              testID="search"
              value=""
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "width": 300,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "width": 176,
            },
            Array [
              Object {
                "width": "100%",
              },
              undefined,
              undefined,
            ],
            Object {
              "opacity": 1,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#D9D9D9",
                "borderRadius": 4,
                "borderWidth": 1,
              },
              Object {
                "width": "100%",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
                "minHeight": 40,
                "overflow": "hidden",
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          >
            <View
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "left": 10,
                  "opacity": 1,
                  "position": "absolute",
                  "right": "auto",
                  "top": "50%",
                  "transform": Array [
                    Object {
                      "translateY": -12.5,
                    },
                  ],
                  "zIndex": 5,
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#999",
                      "fontSize": 25,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <TextInput
              editable={true}
              maxResizableWidth={750}
              minHeight={60}
              minWidth={0}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              paddingLeft={40}
              paddingRight={8}
              placeholder="Search product name"
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              selectionColor="#0768FD"
              style={
                Array [
                  Object {
                    "margin": 0,
                    "textAlign": "left",
                    "zIndex": 5,
                  },
                  undefined,
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "height": 40,
                    "minHeight": 0,
                    "paddingLeft": 40,
                    "paddingRight": 8,
                    "paddingVertical": 5,
                    "textAlignVertical": "center",
                    "width": "100%",
                  },
                ]
              }
              testID="search-view"
              value=""
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "width": 300,
        }
      }
      testID="filterWrapper"
    >
      <View
        style={
          Array [
            Object {
              "width": "100%",
            },
            undefined,
          ]
        }
      >
        <View>
          <View
            collapsable={false}
          >
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Object {
                    "backgroundColor": "rgba(246, 246, 246, 0.4)",
                    "borderColor": "rgba(217, 217, 217, 0.4)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "elevation": 0,
                    "height": 40,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "stretch",
                    "flex": 1,
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "paddingLeft": 8,
                  }
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  hitSlop={
                    Object {
                      "bottom": 14,
                      "left": 14,
                      "right": 14,
                      "top": 14,
                    }
                  }
                  nativeID="animatedComponent"
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "color": "rgba(0, 0, 0, 0.4)",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": "400",
                        },
                      ]
                    }
                  >
                    Brand
                  </Text>
                </View>
              </View>
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": true,
                  }
                }
                accessible={true}
                centered={true}
                collapsable={false}
                focusable={true}
                hitSlop={
                  Object {
                    "bottom": 6,
                    "left": 6,
                    "right": 6,
                    "top": 6,
                  }
                }
                nativeID="animatedComponent"
                onClick={[Function]}
                onLayout={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="box-only"
                style={Object {}}
              >
                <View
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "margin": 6,
                          "overflow": "hidden",
                        },
                        Object {
                          "borderRadius": 18,
                          "height": 36,
                          "width": 36,
                        },
                        Object {
                          "opacity": 0.32,
                        },
                        Object {
                          "margin": 0,
                        },
                      ],
                    ]
                  }
                >
                  <View>
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      style={
                        Array [
                          Object {
                            "color": "rgba(0, 0, 0, 0.4)",
                            "fontSize": 24,
                          },
                          Array [
                            Object {
                              "transform": Array [
                                Object {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                          ],
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 0,
                        "left": 0,
                        "overflow": "hidden",
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "borderRadius": 0,
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
